### ISSUE TITLE
AV_NOPTS_VALUE removed

### ISSUE THREAD

**Issue by ispysoftware** [Ruslan-B/FFmpeg.AutoGen#55]
Any reason?

**Comment by Ruslan-B** [comment]
Context?

**Comment by ispysoftware** [comment]
in **public unsafe struct AVPacket**:

/// <summary>Presentation timestamp in AVStream-&gt;time_base units; the time at which the decompressed packet will be presented to the user. Can be AV_NOPTS_VALUE if it is not stored in the file. pts MUST be larger or equal to dts as presentation cannot happen before decompression, unless one wants to view hex dumps. Some formats misuse the terms dts and pts/cts to mean something different. Such timestamps must be converted to true pts/dts before they are stored in AVPacket.</summary>
        public long @pts;

- cant set it to AV_NOPTS_VALUE as AV_NOPTS_VALUE doesn't exist anymore

**Comment by Ruslan-B** [comment]
It is in place actually as malformed macro:
       // public static  AV_NOPTS_VALUE = ((int64_t)UINT64_C(0x8000000000000000));
My macro parser is quite limed. Thus at the moment you need to define this constant yourself.

**Comment by ispysoftware** [comment]
yes is saw that, uncommenting it doesn't work as int64_t can't be resolved - any idea how to set it?

**Comment by ispysoftware** [comment]
.. it used to be

public const ulong AV_NOPTS_VALUE = 0x8000000000000000;

**Comment by Ruslan-B** [comment]
You know that all this code is auto generated aren't you? Thus in case you would like to have this value here there are two options:
- contribute try to fix in original C header (by some reason they put in place this double cast).
- contribute good generic fix to macro parser to my progect as adding/fixing stuff by hands is not an option.
Beside, you still going to have small amount of macroses based on inline functions they never worked and next time ffmpeg developers desided to use one this constant will be gone.
To summarize I will do it with pleasure when CppSharp project will add macro preprocessor api.

**Comment by ispysoftware** [comment]
Hello yes i know its autogen.. if the autogen isnt working though it's
probably easier to manually fix something like this that is rather critical
to the library. The FFMPEG guys keep making massive changes to the api.
Quite frustrating!

**Comment by Ruslan-B** [comment]
Any way I'll try to improve macro parser but I cannot promise any backward compatibility connected with ffmpeg api changes either this specific constraint being resolved.

**Comment by Ruslan-B** [comment]
I implemented generic fix  for macro functions in #63 thus AV_NOPTS_VALUE is now available.
The nuget package is also updated.
